src/connection.h

```diff
@@ -142,22 +120,20 @@ class Connection : public std::enable_shared_from_this<Connection>
 
 		std::recursive_mutex m_connectionLock;
 
+		std::list<OutputMessage_ptr> messageQueue;
+
 		ServicePort_ptr m_service_port;
+		Protocol_ptr m_protocol;
 
-		boost::asio::ip::tcp::socket* m_socket;
-		Protocol* m_protocol;
+		std::unique_ptr<boost::asio::ip::tcp::socket> m_socket;
 		boost::asio::io_service& m_io_service;
 
 		time_t m_timeConnected;
 		uint32_t m_packetsSent;
-		uint32_t m_refCount;
-		int32_t m_pendingWrite;
-		int32_t m_pendingRead;
-		ConnectionState_t m_connectionState;
 
+		bool m_pendingWrite;
```

Comment:
Are you sure that this can be a boolean? Can there not be more than one pending write at a time?